becquerel/core/autocal.py

```diff
@@ -0,0 +1,267 @@
+"""Automatic calibration by associating peaks with energies."""
+
+from __future__ import print_function
+from itertools import combinations
+import matplotlib.pyplot as plt
+import numpy as np
+from .peakfinder import PeakFinder
+from .energycal import LinearEnergyCal
+
+
+MAJOR_BACKGROUND_LINES = [
+    1460.82,
+    2614.51,
+    1620.50,
+    911.20,
+    338.32,
+    238.63,
+    2204.06,
+    1764.49,
+    1377.67,
+    1238.12,
+    1120.29,
+    768.36,
+    609.32,
+    351.93,
+    295.22,
+    242.00,
+    186.21,
+]
+
+
+def fit_gain(channels, snrs, energies):
+    """Calculate the mean gain to relate channels to energies.
+
+    Assume that the peak channel error scales like fwhm/snr,
+    which is proportional to sqrt(x)/snr.
+
+    Minimize:
+        F = sum(snr^2 * (x * g - y)^2 / x)
+        dF/dg = sum(2 * snr^2 * x * (x * g - y) / x)
+              = 2 * sum(snr^2 * (x * g - y))
+              = 2 * (g * sum(snr^2 * x) - sum(snr^2 * y))
+              = 0
+        g = sum(snr^2 * y) / sum(snr^2 * x)
+
+    """
+    assert len(channels) == len(energies)
+    assert len(channels) == len(snrs)
```

Comment:
@markbandstra can you change these from `assert` to raise an `AutoCalibrationError`?